src/main/java/com/gluonhq/gradle/tasks/ConfigBuild.java

```diff
@@ -61,74 +62,74 @@
+                break;
+            case Constants.PROFILE_LINUX_AARCH64:
+                targetTriplet = new Triplet(Constants.Profile.LINUX_AARCH64);
                 break;
             default:
                 throw new RuntimeException("No valid target found for " + target);
         }
         clientConfig.setTarget(targetTriplet);
 
-        clientConfig.setBackend(clientExtension.getBackend().toLowerCase(Locale.ROOT));
         clientConfig.setBundlesList(clientExtension.getBundlesList());
         clientConfig.setResourcesList(clientExtension.getResourcesList());
-        clientConfig.setDelayInitList(clientExtension.getDelayInitList());
         clientConfig.setJniList(clientExtension.getJniList());
+        clientConfig.setCompilerArgs(clientExtension.getCompilerArgs());
         clientConfig.setReflectionList(clientExtension.getReflectionList());
-        clientConfig.setRuntimeArgsList(clientExtension.getRuntimeArgsList());
-        clientConfig.setReleaseSymbolsList(clientExtension.getReleaseSymbolsList());
-
-        clientConfig.setMainClassName((String) project.getProperties().get("mainClassName"));
+        clientConfig.setAppId(project.getGroup() + "." + project.getName());
         clientConfig.setAppName(project.getName());
 
-        List<Path> classPath = getClassPathFromSourceSets();
-        clientConfig.setUseJavaFX(classPath.stream().anyMatch(f -> f.getFileName().toString().contains("javafx")));
-        clientConfig.setGraalLibsUserPath(clientExtension.getGraalLibsPath());
+        clientConfig.setGraalPath(Path.of(getGraalHome().get()));
 
-        clientConfig.setLlcPath(clientExtension.getLlcPath());
-        clientConfig.setEnableCheckHash(clientExtension.isEnableCheckHash());
-        clientConfig.setUseJNI(clientExtension.isUseJNI());
+        clientConfig.setUsePrismSW(clientExtension.isEnableSwRendering());
         clientConfig.setVerbose(clientExtension.isVerbose());
-    }
 
-    Configuration getClientConfig() {
+        IosSigningConfiguration iosConfiguration = new IosSigningConfiguration();
+        iosConfiguration.setProvidedSigningIdentity(clientExtension.getIosExtension().getSigningIdentity());
+        iosConfiguration.setProvidedProvisioningProfile(clientExtension.getIosExtension().getProvisioningProfile());
+        iosConfiguration.setSimulatorDevice(clientExtension.getIosExtension().getSimulatorDevice());
+        // TODO: Allow frameworks
+        clientConfig.setIosSigningConfiguration(iosConfiguration);
+        
         return clientConfig;
     }
 
     void build() {
+    	ProjectConfiguration clientConfig = createSubstrateConfiguration();
+    	
+        if (!getGraalHome().isPresent()) {
+            throw new GradleException("GraalVM installation directory not found." +
+                    " Either set GRAALVM_HOME as an environment variable or" +
+                    " set graalvmHome in the client-plugin configuration");
```

Comment:
This check should be the first thing in the method. A call is currently made to `createSubstrateConfiguration()` before this check, it [sets GraalPath](https://github.com/gluonhq/client-gradle-plugin/pull/61/files#diff-e9fc26070925cb33b1e898c927e18cd0R108) without checking if its present.

Reply:
Fixed